tests/StorageApiTestCase.php

```diff
@@ -673,7 +685,140 @@ function (self $that, array $config, $useExistingBranch = false) {
         ];
     }
 
-    private function createOrReuseDevBranch(self $that, $useExistingBranch = false)
+    /**
+     * Usage:
+     *  - `$getClient($this)` -> return default client which use branch aware client for calling default branch
+     *  - `$getClient($this, [...])` -> return client with custom config which use branch aware client for calling default branch
+     *  - `$getClient($this, [], true)` -> return default client which use same branch as before
+     *  - `$getClient($this, [...], true)` -> return client with custom which use same branch as before
+     *
+     * @see provideComponentsClient
+     */
+    public function provideBranchAwareComponentsClient()
+    {
+        yield 'defaultBranch' => [
+            function (self $that, array $config = []) {
+                if ($config) {
+                    return $this->getBranchAwareClient($this->getDefaultBranchId($that), $config);
+                } else {
+                    return $this->getBranchAwareDefaultClient($this->getDefaultBranchId($that));
+                }
+            }
+        ];
+
+        yield 'devBranch' => [
+            function (self $that, array $config = [], $useExistingBranch = false) {
+                $branch = $this->createOrReuseDevBranch($that, $useExistingBranch);
+
+                if ($config) {
+                    return $this->getBranchAwareClient($branch['id'], $config);
+                } else {
+                    return $this->getBranchAwareDefaultClient($branch['id']);
+                }
+            }
+        ];
+    }
+
+    protected function getDefaultBranchId(self $that)
+    {
+        $devBranch = new \Keboola\StorageApi\DevBranches($that->_client);
+        $branchesList = $devBranch->listBranches();
+        foreach ($branchesList as $branch) {
+            if ($branch['isDefault'] === true) {
+                return $branch['id'];
+            }
+        }
+
+        return null;
+    }
+
+    public function cleanupConfigurations()
+    {
+        $components = new Components($this->_client);
+        foreach ($components->listComponents() as $component) {
+            foreach ($component['configurations'] as $configuration) {
+                $components->deleteConfiguration($component['id'], $configuration['id']);
+            }
+        }
+
+        // erase all deleted configurations
+        foreach ($components->listComponents((new ListComponentsOptions())->setIsDeleted(true)) as $component) {
+            foreach ($component['configurations'] as $configuration) {
+                $components->deleteConfiguration($component['id'], $configuration['id']);
+            }
+        }
+    }
+
+    /**
+     * @param string $eventName
+     * @return array
+     */
+    protected function listEvents(Client $client, $eventName, $expectedObjectId = null)
+    {
+        return $this->retry(function () use ($client, $expectedObjectId) {
```

Comment:
Škoda, že tady není Keboola\Retry :( Tenhle retry se chová úplně jinak, než ten standadní...

Reply:
Ale zas to tím tady asi nemusíme brzdit, když to funguje